packages/protocol/contracts/rrp/authorizers/admin/AirnodeTokenLock.sol

```diff
@@ -0,0 +1,433 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.6;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "../../../authorizers/interfaces/IRequesterAuthorizerWithManager.sol";
+import "./AirnodeTokenLockRolesWithManager.sol";
+import "./interfaces/IAirnodeFeeRegistry.sol";
+import "./interfaces/IAirnodeTokenLock.sol";
+
+/// @title The contract used to lock API3 Tokens in order to gain access to Airnodes
+contract AirnodeTokenLock is
+    AirnodeTokenLockRolesWithManager,
+    IAirnodeTokenLock
+{
+    string private constant ERROR_ZERO_CHAINID = "Zero ChainID";
+    string private constant ERROR_ZERO_ADDRESS = "Zero address";
+    string private constant ERROR_NOT_AIRNODE = "Caller not Airnode";
+    string private constant ERROR_ZERO_AMOUNT = "Zero amount";
+    string private constant ERROR_INSUFFICIENT_AMOUNT = "Insufficient amount";
+    string private constant ERROR_ALREADY_LOCKED = "Already locked";
+    string private constant ERROR_NOT_LOCKED = "No amount locked";
+    string private constant ERROR_LOCK_PERIOD_NOT_EXPIRED =
+        "Locking period not expired";
+    string private constant ERROR_REQUESTER_BLOCKED = "Requester Blocked";
+    string private constant ERROR_REQUESTER_NOT_BLOCKED =
+        "Requester Not Blocked";
+    string private constant ERROR_AIRNODE_NOT_OPTED_IN = "Airnode not opted in";
+    string private constant ERROR_NOT_ORACLE = "Not oracle";
+    string private constant ERROR_NOT_ORACLE_ADDRESS_SETTER =
+        "Not oracle address setter";
+    string private constant ERROR_NOT_OPT_STATUS_SETTER =
+        "Not opt status setter";
+    string private constant ERROR_NOT_BLOCK_WITHDRAW_DESTINATION_SETTER =
+        "Not block withdraw destination setter";
+    string private constant ERROR_NOT_BLOCK_REQUESTER = "Not block requester";
+    string private constant ERROR_NOT_REQUESTER_AUTHORIZER_WITH_MANAGER_SETTER =
+        "Not RequesterAuthorizerWithManagerSetter";
+
+    /// @dev Address of Api3Token
+    address public immutable api3Token;
+
+    /// @dev Address of AirnodeFeeRegistry
+    address public immutable airnodeFeeRegistry;
+
+    /// @dev The price of API3 in terms of USD
+    /// @notice The price is specified upto 6 decimal places
+    uint256 public api3PriceInUsd;
+
+    /// @dev A coefficient used to calculate the amount of tokens to be locked
+    uint8 public multiplierCoefficient;
```

Comment:
Since `multiplierCoefficient` is type `uint8`, shouldn't we move above `api3PriceInUsd` to try pack it in a single memory slot with `airnodeFeeRegistry` (20bytes)?

Reply:
I've changed it to be a uint256, mainly because it didn't make sense to have it be uint8 integer. We would want precision in the coefficient when setting it, before it could only be set without precision. Now it can be set as a number with 18 digit decimals which also makes the calculation of the tokenlock cleaner, all the numbers are in their respective decimals and we don't have to multiply/divide by anything.